api/enki.py

```diff
@@ -0,0 +1,542 @@
+            # make _make_request raise a RemoteIntegrationException.
+            #
+            # The token has expired. Get a new token and try again.
+            self.token = None
+            return self.request(
+                url=url, method=method, extra_headers=extra_headers,
+                data=data, params=params, exception_on_401=True
+            )
+        else:
+            return response
+
+    def availability(self, patron_id=None, since=None, title_ids=[], strt=0, qty=2000):
+        self.log.debug ("requesting : "+ str(qty) + " books starting at econtentRecord" +  str(strt))
+        url = str(self.base_url) + str(self.availability_endpoint)
+        args = dict()
+	args['method'] = "getAllTitles"
+	args['id'] = "secontent"
+        args['strt'] = strt
+        args['qty'] = qty
+        args['lib'] = self.lib
+	response = self.request(url, params=args)
+        return response
+
+    @classmethod
+    def create_identifier_strings(cls, identifiers):
+        identifier_strings = []
+        for i in identifiers:
+            if isinstance(i, Identifier):
+                value = i.identifier
+            else:
+                value = i
+            identifier_strings.append(value)
+
+        return identifier_strings
+
+    @classmethod
+    def parse_token(cls, token):
+        data = json.loads(token)
+        return data['access_token']
+
+    def _make_request(self, url, method, headers, data=None, params=None,
+                      **kwargs):
+        """Actually make an HTTP request."""
+        return HTTP.request_with_timeout(
+            method, url, headers=headers, data=data,
+            params=params, **kwargs
+        )
+
+    def reaper_request(self, identifier):
+        self.log.debug ("Checking availability for " + str(identifier.identifier))
+        now = datetime.datetime.utcnow()
+        url = str(self.base_url) + str(self.item_endpoint)
+        args = dict()
+        args['method'] = "getItem"
+        args['recordid'] = identifier.identifier
+        args['size'] = "small"
+        args['lib'] = self.lib
+        response = self.request(url, method='get', params=args)
+        if not(response.content.startswith("{\"result\":{\"id\":\"")):
+            pool = identifier.licensed_through[0]
+            if pool and (pool.licenses_owned > 0):
+                if pool.presentation_edition:
+                    self.log.warn("Removing %s (%s) from circulation",
+                                  pool.presentation_edition.title, pool.presentation_edition.author)
+                else:
+                    self.log.warn(
+                        "Removing unknown work %s from circulation.",
+                        identifier.identifier
+                    )
+            pool.licenses_owned = 0
+            pool.licenses_available = 0
+            pool.licenses_reserved = 0
+            pool.patrons_in_hold_queue = 0
+            pool.last_checked = now
```

Comment:
Calling CirculationData.apply() should set these values, so you shouldn't need to set them inside this method.